src/main.py

```diff
@@ -112,6 +112,28 @@ def main(args):
     elif os.path.isfile(os.path.join(ckpt_path, "summary.json")): # the experiment was already completed
         print(f"Already found experiment '{ckpt_path}'.\nSkipping.")
         sys.exit(0)
+    itr = 0
+    rng_state_dict = None
+    distributed_backend.sync()
```

Comment:
can you put this inside the if not os.path.exists(ckpt_path)? it makes it more clear why it's needed.

Reply:
I only moved   ` distributed_backend.sync()` inside as the other variables need to be defined if the path already exists but there are no checkpoints which might happen in case of crashing early on

Reply:
yes that's exactly what I meant. Thank you.